### DISCUSSION TITLE
Moes Electric Floor Heater BHT-002-GBLW

### DISCUSSION THREAD

**Discussion by hillaliy** [RavenSystem/esp-homekit-devices#1834] [MEPLHAA Scripts Collection]
![image](https://user-images.githubusercontent.com/950010/170618002-0335beba-e174-4e74-a370-7828a3e667e7.png)

{"c":{"r":[{"n":0,"s":9600}],"ul":[8,50],"ntp":"pool.ntp.org","tz":"UTC+2"},"a":[{"t":21,"w":1,"m":5,"x":35,"st":0.5,"s":5,"0":{"u":[{"v":"55aa0006000501010001000d"}]},"5":{"m":[[0,0.02]]},"14":{"u":[{"v":"55aa0006000501010001010e"}]},"y2":[{"v":5,"0":{"u":[{"v":"55aa00060008020200040000000a1f"}]}},{"v":5.5,"0":{"u":[{"v":"55aa00060008020200040000000b20"}]}},{"v":6,"0":{"u":[{"v":"55aa00060008020200040000000c21"}]}},{"v":6.5,"0":{"u":[{"v":"55aa00060008020200040000000d22"}]}},{"v":7,"0":{"u":[{"v":"55aa00060008020200040000000e23"}]}},{"v":7.5,"0":{"u":[{"v":"55aa00060008020200040000000f24"}]}},{"v":8,"0":{"u":[{"v":"55aa00060008020200040000001025"}]}},{"v":8.5,"0":{"u":[{"v":"55aa00060008020200040000001126"}]}},{"v":9,"0":{"u":[{"v":"55aa00060008020200040000001227"}]}},{"v":9.5,"0":{"u":[{"v":"55aa00060008020200040000001328"}]}},{"v":10,"0":{"u":[{"v":"55aa00060008020200040000001429"}]}},{"v":10.5,"0":{"u":[{"v":"55aa0006000802020004000000152a"}]}},{"v":11,"0":{"u":[{"v":"55aa0006000802020004000000162b"}]}},{"v":11.5,"0":{"u":[{"v":"55aa0006000802020004000000172c"}]}},{"v":12,"0":{"u":[{"v":"55aa0006000802020004000000182d"}]}},{"v":12.5,"0":{"u":[{"v":"55aa0006000802020004000000192e"}]}},{"v":13,"0":{"u":[{"v":"55aa00060008020200040000001a2f"}]}},{"v":13.5,"0":{"u":[{"v":"55aa00060008020200040000001b30"}]}},{"v":14,"0":{"u":[{"v":"55aa00060008020200040000001c31"}]}},{"v":14.5,"0":{"u":[{"v":"55aa00060008020200040000001d32"}]}},{"v":15,"0":{"u":[{"v":"55aa00060008020200040000001e33"}]}},{"v":15.5,"0":{"u":[{"v":"55aa00060008020200040000001f34"}]}},{"v":16,"0":{"u":[{"v":"55aa00060008020200040000002035"}]}},{"v":16.5,"0":{"u":[{"v":"55aa00060008020200040000002136"}]}},{"v":17,"0":{"u":[{"v":"55aa00060008020200040000002237"}]}},{"v":17.5,"0":{"u":[{"v":"55aa00060008020200040000002338"}]}},{"v":18,"0":{"u":[{"v":"55aa00060008020200040000002439"}]}},{"v":18.5,"0":{"u":[{"v":"55aa0006000802020004000000253a"}]}},{"v":19,"0":{"u":[{"v":"55aa0006000802020004000000263b"}]}},{"v":19.5,"0":{"u":[{"v":"55aa0006000802020004000000273c"}]}},{"v":20,"0":{"u":[{"v":"55aa0006000802020004000000283d"}]}},{"v":20.5,"0":{"u":[{"v":"55aa0006000802020004000000293e"}]}},{"v":21,"0":{"u":[{"v":"55aa00060008020200040000002a3f"}]}},{"v":21.5,"0":{"u":[{"v":"55aa00060008020200040000002b40"}]}},{"v":22,"0":{"u":[{"v":"55aa00060008020200040000002c41"}]}},{"v":22.5,"0":{"u":[{"v":"55aa00060008020200040000002d42"}]}},{"v":23,"0":{"u":[{"v":"55aa00060008020200040000002e43"}]}},{"v":23.5,"0":{"u":[{"v":"55aa00060008020200040000002f44"}]}},{"v":24,"0":{"u":[{"v":"55aa00060008020200040000003045"}]}},{"v":24.5,"0":{"u":[{"v":"55aa00060008020200040000003146"}]}},{"v":25,"0":{"u":[{"v":"55aa00060008020200040000003247"}]}},{"v":25.5,"0":{"u":[{"v":"55aa00060008020200040000003348"}]}},{"v":26,"0":{"u":[{"v":"55aa00060008020200040000003449"}]}},{"v":26.5,"0":{"u":[{"v":"55aa0006000802020004000000354a"}]}},{"v":27,"0":{"u":[{"v":"55aa0006000802020004000000364b"}]}},{"v":27.5,"0":{"u":[{"v":"55aa0006000802020004000000374c"}]}},{"v":28,"0":{"u":[{"v":"55aa0006000802020004000000384d"}]}},{"v":28.5,"0":{"u":[{"v":"55aa0006000802020004000000394e"}]}},{"v":29,"0":{"u":[{"v":"55aa00060008020200040000003a4f"}]}},{"v":29.5,"0":{"u":[{"v":"55aa00060008020200040000003b50"}]}},{"v":30,"0":{"u":[{"v":"55aa00060008020200040000003c51"}]}},{"v":30.5,"0":{"u":[{"v":"55aa00060008020200040000003d52"}]}},{"v":31,"0":{"u":[{"v":"55aa00060008020200040000003e53"}]}},{"v":31.5,"0":{"u":[{"v":"55aa00060008020200040000003f54"}]}},{"v":32,"0":{"u":[{"v":"55aa00060008020200040000004055"}]}},{"v":32.5,"0":{"u":[{"v":"55aa00060008020200040000004156"}]}},{"v":33,"0":{"u":[{"v":"55aa00060008020200040000004257"}]}},{"v":33.5,"0":{"u":[{"v":"55aa00060008020200040000004358"}]}},{"v":34,"0":{"u":[{"v":"55aa00060008020200040000004459"}]}},{"v":34.5,"0":{"u":[{"v":"55aa0006000802020004000000455a"}]}},{"v":35,"0":{"u":[{"v":"55aa0006000802020004000000465b"}]}}],"es":[{"t":22},{"t":80,"n":24,"dt":[1,0],"pt":[["55AA0107000803020004000000"]],"ff":0.5,"tg":[1,0]},{"t":80,"n":24,"dt":[1,0],"pt":[["55aa01070005010100010"]],"tg":[1,2]},{"t":80,"n":24,"dt":[1,0],"pt":[["55aa0107000802020004000000"]],"ff":0.5,"tg":[1,5]},{"t":80,"n":24,"dt":[1,0],"pt":[["55AA0107000866020004000000"]],"ff":0.5,"tg":[2,0]}]}]}

**Comment by gunoid** [comment]
@hillaliy Could you pls share your experience with Moes floor heater.
Did you flash HAA firmware over Tuya original with this json? How does it work with Homekit?

**Reply by hillaliy** [reply]
Hi, Tuya Convert didn't work for me. So I ditched the TYWE3S chip, Then flash it with HAA and reconnect. I have a soldering gun.
It's work perfect with HAA.

**Reply by jmgutierrezc1** [reply]
Hi, this is a great add to the community, maybe considering the project but no clearly explained. Did you change the tuya chip for an esp8266, when you say "I dicthed the TYE..", or you just reflashed reflashed it with HAA?.
Much appreciated!!

**Reply by borissov** [reply]
> Hi, this is a great add to the community, maybe considering the project but no clearly explained. Did you change the tuya chip for an esp8266, when you say "I dicthed the TYE..", or you just reflashed reflashed it with HAA?.
> 
> Much appreciated!!

Nope, the chip inside is esp8266. But flashing process wakes up and other chips on board and is a bit tricky.

**Reply by jmgutierrezc1** [reply]
could not flash the chip at all, never established connection with it, not even when removed from the PCB. The chip is working fine as I reinstalled with the original software and is working with the MOES app. Any ideas?

Thanks a lot

**Reply by jmgutierrezc1** [reply]
Thank you very much for this project I managed to flash a new esp12 and replace the orignal chip, my only problem is that the temperature reading is not working. i see in home app -99 degrees and not changing. the temperature control, on/off work perfectly. Any ideas??

**Reply by jmgutierrezc1** [reply]
Actually is the whole commmuncation back to the home app that is not working. When changin manually the temp set in the device, home app does not react.

**Reply by RavenSystem** [reply]
Are you using the updated script? https://github.com/RavenSystem/esp-homekit-devices/discussions/1834#discussioncomment-6078025

**Reply by jmgutierrezc1** [reply]
checking but I think so, I will reload the chip with script with this one thanks. Will let you know

**Reply by jmgutierrezc1** [reply]
relaoded the script and still no answer. already verified that RX and TX have solid connection to PCB...
![Captura de pantalla 2024-06-13 a la(s) 11 43 20 a m](https://github.com/RavenSystem/esp-homekit-devices/assets/45322970/48fd9cb2-9eea-424d-8bf1-9e2a217fcf7a)

**Reply by jmgutierrezc1** [reply]
![IMG_1199](https://github.com/RavenSystem/esp-homekit-devices/assets/45322970/7b7def77-ba9c-44e6-93f7-c65320cfcb46)
evrything looks ok but still do not receving any messages from the MCU to the ESP

**Reply by jmgutierrezc1** [reply]
Any aideas to fix this are welcome, thank you all!

**Reply by jmgutierrezc1** [reply]
Sorry typos….

**Reply by jmgutierrezc1** [reply]
Sorry guys, some more information. Just found that the chip that Came in the thermostat is not a TYWE3S, which is ES8266 based, but a WB3S, BK7231T based., This is the reason I could not flash this chip and probably why the messages coming from the thermostat (temp. reading and manual changes in the thermostat) are NOT coming back to homekit. Thoughts?

**Reply by hillaliy** [reply]
Maybe Tuya changed the serial commands. You need to check it.

**Reply by jmgutierrezc1** [reply]
Totally unfamiliar with configuration of UART messages, would not know where to start. Unfortunately I already have the esp replacing the other chip and half way there. Can someone recommend where to get information.

**Reply by hillaliy** [reply]
You won't find it on the net. If I have one of this switch model I can get the serial commands.

**Reply by jmgutierrezc1** [reply]
What tools/process do you recommend for that?. Thanks

**Reply by TriderG75** [reply]
> Sorry guys, some more information. Just found that the chip that Came in the thermostat is not a TYWE3S, which is ES8266 based, but a WB3S, BK7231T based., This is the reason I could not flash this chip and probably why the messages coming from the thermostat (temp. reading and manual changes in the thermostat) are NOT coming back to homekit. Thoughts?

Found the right commands. I have a working script to test if someone needs it.

**Reply by jmgutierrezc1** [reply]
I need it pelase

**Reply by TriderG75** [reply]
> I need it pelase

There is also a switch to lock/unlock the device. I limited the device to 26.5 degrees. It has one temperature sensor. Let me know if it works for you

{"c":{"r":[{"n":10,"s":9600}],"ntp":"pool.ntp.org","tz":"UTC+2","1":{"u":[{"v":"55aa002b0001042f"},{"v":"55aa001c00080119030e161f2805b0"},{"v":"55aa000300010407"}]}},"a":[{"t":21,"m":5,"x":26.5,"st":0.5,"s":5,"0":{"u":[{"v":"55aa0006000501010001000d"}]},"5":{"m":[[0,0.02]]},"14":{"u":[{"v":"55aa0006000501010001010e"}]},"y2":[{"v":5,"0":{"u":[{"v":"55aa00060008020200040000000a1f"}]}},{"v":5.5,"0":{"u":[{"v":"55aa00060008020200040000000b20"}]}},{"v":6,"0":{"u":[{"v":"55aa00060008020200040000000c21"}]}},{"v":6.5,"0":{"u":[{"v":"55aa00060008020200040000000d22"}]}},{"v":7,"0":{"u":[{"v":"55aa00060008020200040000000e23"}]}},{"v":7.5,"0":{"u":[{"v":"55aa00060008020200040000000f24"}]}},{"v":8,"0":{"u":[{"v":"55aa00060008020200040000001025"}]}},{"v":8.5,"0":{"u":[{"v":"55aa00060008020200040000001126"}]}},{"v":9,"0":{"u":[{"v":"55aa00060008020200040000001227"}]}},{"v":9.5,"0":{"u":[{"v":"55aa00060008020200040000001328"}]}},{"v":10,"0":{"u":[{"v":"55aa00060008020200040000001429"}]}},{"v":10.5,"0":{"u":[{"v":"55aa0006000802020004000000152a"}]}},{"v":11,"0":{"u":[{"v":"55aa0006000802020004000000162b"}]}},{"v":11.5,"0":{"u":[{"v":"55aa0006000802020004000000172c"}]}},{"v":12,"0":{"u":[{"v":"55aa0006000802020004000000182d"}]}},{"v":12.5,"0":{"u":[{"v":"55aa0006000802020004000000192e"}]}},{"v":13,"0":{"u":[{"v":"55aa00060008020200040000001a2f"}]}},{"v":13.5,"0":{"u":[{"v":"55aa00060008020200040000001b30"}]}},{"v":14,"0":{"u":[{"v":"55aa00060008020200040000001c31"}]}},{"v":14.5,"0":{"u":[{"v":"55aa00060008020200040000001d32"}]}},{"v":15,"0":{"u":[{"v":"55aa00060008020200040000001e33"}]}},{"v":15.5,"0":{"u":[{"v":"55aa00060008020200040000001f34"}]}},{"v":16,"0":{"u":[{"v":"55aa00060008020200040000002035"}]}},{"v":16.5,"0":{"u":[{"v":"55aa00060008020200040000002136"}]}},{"v":17,"0":{"u":[{"v":"55aa00060008020200040000002237"}]}},{"v":17.5,"0":{"u":[{"v":"55aa00060008020200040000002338"}]}},{"v":18,"0":{"u":[{"v":"55aa00060008020200040000002439"}]}},{"v":18.5,"0":{"u":[{"v":"55aa0006000802020004000000253a"}]}},{"v":19,"0":{"u":[{"v":"55aa0006000802020004000000263b"}]}},{"v":19.5,"0":{"u":[{"v":"55aa0006000802020004000000273c"}]}},{"v":20,"0":{"u":[{"v":"55aa0006000802020004000000283d"}]}},{"v":20.5,"0":{"u":[{"v":"55aa0006000802020004000000293e"}]}},{"v":21,"0":{"u":[{"v":"55aa00060008020200040000002a3f"}]}},{"v":21.5,"0":{"u":[{"v":"55aa00060008020200040000002b40"}]}},{"v":22,"0":{"u":[{"v":"55aa00060008020200040000002c41"}]}},{"v":22.5,"0":{"u":[{"v":"55aa00060008020200040000002d42"}]}},{"v":23,"0":{"u":[{"v":"55aa00060008020200040000002e43"}]}},{"v":23.5,"0":{"u":[{"v":"55aa00060008020200040000002f44"}]}},{"v":24,"0":{"u":[{"v":"55aa00060008020200040000003045"}]}},{"v":24.5,"0":{"u":[{"v":"55aa00060008020200040000003146"}]}},{"v":25,"0":{"u":[{"v":"55aa00060008020200040000003247"}]}},{"v":25.5,"0":{"u":[{"v":"55aa00060008020200040000003348"}]}},{"v":26,"0":{"u":[{"v":"55aa00060008020200040000003449"}]}},{"v":26.5,"0":{"u":[{"v":"55aa0006000802020004000000354a"}]}}],"es":[
{"t":80,"n":24,"dt":[1,0],"pt":[["55aa0307000803020004000000"]],"ff":0.5,"tg":[1,0]},{"t":80,"n":24,"dt":[1,0],"pt":[["55aa03070005010100010"]],"tg":[1,2]},
{"t":80,"n":24,"dt":[1,0],"pt":[["55aa0307000802020004000000"]],"ff":0.5,"tg":[1,5]},
{"0":{"u":[{"v":"55aa00060005060100010012"}]},"1":{"u":[{"v":"55aa00060005060100010113"}]}}]}]}

**Reply by jmgutierrezc1** [reply]
Thanks a lot!!!. Will try and lee you know..

**Reply by jmgutierrezc1** [reply]
Hi thanks for the help. Sorry to report that still not working for me. The communication back to HomeKit not established. When moving temperature manually in the control, HomeKit does not notice it and the temp is not being reported back to HomeKit.

**Reply by jmgutierrezc1** [reply]
![IMG_2338](https://github.com/user-attachments/assets/0324cece-8ab1-4926-9b70-3e7b8030ae4f)

**Reply by TriderG75** [reply]
Tried right now. Removed and paired again with the script I sent you.

**Reply by TriderG75** [reply]
![IMG_058008306E4D-1](https://github.com/user-attachments/assets/01f832f4-c9f6-4081-85c5-7837050bf6c8)

**Reply by jmgutierrezc1** [reply]
Not clear what is happening. I replaced the original chip with an esp8266 12. Not sure that may cause the lack of communication but sure that it is not working even after re-pairing it. Thanks for your help

**Reply by TriderG75** [reply]
Still not working? I'm used it for a few day and it's ok. Are you on the haa channel on Discord?

**Comment by GPL71** [comment]
Is display working with HAA?
What this MEPLHAA do?
Thank you

**Reply by hillaliy** [reply]
The buttons and the LCD control by the MCU chip, so it's work with HAA

**Comment by pilot1981** [comment]
@hillaliy are you sure you are using BHT-002-GB? It don’t have WIFI connection?!

**Reply by hillaliy** [reply]
Hi this is what I'm using

₪ 96.22  45％ Off | WiFi Thermostat Temperature Controller LCD Touch Screen Backlight for Electric Heating Works with Alexa Google Home 16A
https://a.aliexpress.com/_EuL9vIJ

**Comment by GPL71** [comment]
Hello ...
Is it possible modify steps from 0,5 to 0,1?
Although manually display could show 0.5 steps, HAA could send a better regulation
Another question
Is it possible set in MEPLHAA "Z": for offset?

**Reply by hillaliy** [reply]
you can change the MEPLHAA as you want

**Reply by GPL71** [reply]
> you can change the MEPLHAA as you want

Ok but how to?
55aa0006000802020004000000     1c31-->it's 14.0
55aa0006000802020004000000     1d32-->it's 14.5

What is for 14.1, 14.2, 14.3, 14.4?
And does BHT accept that steps? Or it know only 14.0, 14.5, 15 ...?

And what is this code? 55aa00060008020200040000001f34
Is it something likes a HAA IR capture?

thank you

**Comment by hillaliy** [comment]
The device is not support x.1, x.2 ....
this (55aa00060008020200040000001f34) is UART command

A lot of people don't understand the UART protocol. I think to make a live stream on Youtube to explain this.

**Reply by GPL71** [reply]
Ok .... then even insert the offset "Z" will not works

**Comment by hillaliy** [comment]
![Tuya MCU](https://user-images.githubusercontent.com/950010/196881618-69b9801f-53f6-4328-9927-de8871383148.png)

**Reply by GPL71** [reply]
thank you

**Comment by hillaliy** [comment]
The MCU control everything you can't change the program on the MCU.
When you flash HAA it's on the TYWE3S chip. it's send and receive commands from and to the MCU.
The hex command - 55aa00060008020200040000001f34
So you need to fit HAA to the MCU program

**Reply by GPL71** [reply]
yes ... I understood that. I thought was possible send a better temperature command via UART
Thank you

**Reply by Bolshoylu** [reply]
Can you convert this script for new firmware version 12 merlin?

**Comment by hillaliy** [comment]
{"c":{"r":[{"n":10,"s":9600}]

**Reply by Bolshoylu** [reply]
Thanks a lot

**Comment by borissov** [comment]
I flash Moes with HAA and this config but the clock always starts from 00:00 and in HomeKit shows one additional thermometer always on 0 degrees..Any ideas what isn’t working?

Thermostat is BHT-002-GCLW.

<img width="414" alt="image" src="https://github.com/RavenSystem/esp-homekit-devices/assets/2177740/3e66378c-ce99-406e-bef9-0aadd6e78eac">

**Comment by RavenSystem** [comment]
HAA V12 Merlin complete script:

```{"c":{"r":[{"n":10,"s":9600,"l":[8,50]}]},"a":[{"t":21,"w":1,"m":5,"x":35,"st":0.5,"s":5,"0":{"u":[{"v":"55aa0006000501010001000d"}]},"5":{"m":[[0,0.02]]},"14":{"u":[{"v":"55aa0006000501010001010e"}]},"y2":[{"v":5,"0":{"u":[{"v":"55aa00060008020200040000000a1f"}]}},{"v":5.5,"0":{"u":[{"v":"55aa00060008020200040000000b20"}]}},{"v":6,"0":{"u":[{"v":"55aa00060008020200040000000c21"}]}},{"v":6.5,"0":{"u":[{"v":"55aa00060008020200040000000d22"}]}},{"v":7,"0":{"u":[{"v":"55aa00060008020200040000000e23"}]}},{"v":7.5,"0":{"u":[{"v":"55aa00060008020200040000000f24"}]}},{"v":8,"0":{"u":[{"v":"55aa00060008020200040000001025"}]}},{"v":8.5,"0":{"u":[{"v":"55aa00060008020200040000001126"}]}},{"v":9,"0":{"u":[{"v":"55aa00060008020200040000001227"}]}},{"v":9.5,"0":{"u":[{"v":"55aa00060008020200040000001328"}]}},{"v":10,"0":{"u":[{"v":"55aa00060008020200040000001429"}]}},{"v":10.5,"0":{"u":[{"v":"55aa0006000802020004000000152a"}]}},{"v":11,"0":{"u":[{"v":"55aa0006000802020004000000162b"}]}},{"v":11.5,"0":{"u":[{"v":"55aa0006000802020004000000172c"}]}},{"v":12,"0":{"u":[{"v":"55aa0006000802020004000000182d"}]}},{"v":12.5,"0":{"u":[{"v":"55aa0006000802020004000000192e"}]}},{"v":13,"0":{"u":[{"v":"55aa00060008020200040000001a2f"}]}},{"v":13.5,"0":{"u":[{"v":"55aa00060008020200040000001b30"}]}},{"v":14,"0":{"u":[{"v":"55aa00060008020200040000001c31"}]}},{"v":14.5,"0":{"u":[{"v":"55aa00060008020200040000001d32"}]}},{"v":15,"0":{"u":[{"v":"55aa00060008020200040000001e33"}]}},{"v":15.5,"0":{"u":[{"v":"55aa00060008020200040000001f34"}]}},{"v":16,"0":{"u":[{"v":"55aa00060008020200040000002035"}]}},{"v":16.5,"0":{"u":[{"v":"55aa00060008020200040000002136"}]}},{"v":17,"0":{"u":[{"v":"55aa00060008020200040000002237"}]}},{"v":17.5,"0":{"u":[{"v":"55aa00060008020200040000002338"}]}},{"v":18,"0":{"u":[{"v":"55aa00060008020200040000002439"}]}},{"v":18.5,"0":{"u":[{"v":"55aa0006000802020004000000253a"}]}},{"v":19,"0":{"u":[{"v":"55aa0006000802020004000000263b"}]}},{"v":19.5,"0":{"u":[{"v":"55aa0006000802020004000000273c"}]}},{"v":20,"0":{"u":[{"v":"55aa0006000802020004000000283d"}]}},{"v":20.5,"0":{"u":[{"v":"55aa0006000802020004000000293e"}]}},{"v":21,"0":{"u":[{"v":"55aa00060008020200040000002a3f"}]}},{"v":21.5,"0":{"u":[{"v":"55aa00060008020200040000002b40"}]}},{"v":22,"0":{"u":[{"v":"55aa00060008020200040000002c41"}]}},{"v":22.5,"0":{"u":[{"v":"55aa00060008020200040000002d42"}]}},{"v":23,"0":{"u":[{"v":"55aa00060008020200040000002e43"}]}},{"v":23.5,"0":{"u":[{"v":"55aa00060008020200040000002f44"}]}},{"v":24,"0":{"u":[{"v":"55aa00060008020200040000003045"}]}},{"v":24.5,"0":{"u":[{"v":"55aa00060008020200040000003146"}]}},{"v":25,"0":{"u":[{"v":"55aa00060008020200040000003247"}]}},{"v":25.5,"0":{"u":[{"v":"55aa00060008020200040000003348"}]}},{"v":26,"0":{"u":[{"v":"55aa00060008020200040000003449"}]}},{"v":26.5,"0":{"u":[{"v":"55aa0006000802020004000000354a"}]}},{"v":27,"0":{"u":[{"v":"55aa0006000802020004000000364b"}]}},{"v":27.5,"0":{"u":[{"v":"55aa0006000802020004000000374c"}]}},{"v":28,"0":{"u":[{"v":"55aa0006000802020004000000384d"}]}},{"v":28.5,"0":{"u":[{"v":"55aa0006000802020004000000394e"}]}},{"v":29,"0":{"u":[{"v":"55aa00060008020200040000003a4f"}]}},{"v":29.5,"0":{"u":[{"v":"55aa00060008020200040000003b50"}]}},{"v":30,"0":{"u":[{"v":"55aa00060008020200040000003c51"}]}},{"v":30.5,"0":{"u":[{"v":"55aa00060008020200040000003d52"}]}},{"v":31,"0":{"u":[{"v":"55aa00060008020200040000003e53"}]}},{"v":31.5,"0":{"u":[{"v":"55aa00060008020200040000003f54"}]}},{"v":32,"0":{"u":[{"v":"55aa00060008020200040000004055"}]}},{"v":32.5,"0":{"u":[{"v":"55aa00060008020200040000004156"}]}},{"v":33,"0":{"u":[{"v":"55aa00060008020200040000004257"}]}},{"v":33.5,"0":{"u":[{"v":"55aa00060008020200040000004358"}]}},{"v":34,"0":{"u":[{"v":"55aa00060008020200040000004459"}]}},{"v":34.5,"0":{"u":[{"v":"55aa0006000802020004000000455a"}]}},{"v":35,"0":{"u":[{"v":"55aa0006000802020004000000465b"}]}}],"es":[{"t":22},{"t":80,"n":24,"dt":[1,0],"pt":[["55AA0107000803020004000000"]],"ff":0.5,"tg":[1,0]},{"t":80,"n":24,"dt":[1,0],"pt":[["55aa01070005010100010"]],"tg":[1,2]},{"t":80,"n":24,"dt":[1,0],"pt":[["55aa0107000802020004000000"]],"ff":0.5,"tg":[1,5]},{"t":80,"n":24,"dt":[1,0],"pt":[["55AA0107000866020004000000"]],"ff":0.5,"tg":[2,0]}]}]}```

**Reply by borissov** [reply]
Stay only in setup mode, and not boot in my case.

**Reply by RavenSystem** [reply]
There was a typo. Now, it is fixed.

**Comment by borissov** [comment]
Is there a way to synchronize the clock using the NTP service? Is it also possible to change to the "manual mode" when switching from HomeKit?

**Comment by borissov** [comment]
Is it possible around 00:00 for the ESP to send a command to the Tuya controller to set its clock to 00:00, so that once a day the ESP's NTP clock synchronizes the Tuya controller's clock?

**Reply by TriderG75** [reply]
Yes it's possible but the problem is the the tuya command includes even day and year...

**Comment by MrJelly17** [comment]
Hi. This is exactly what I was looking for. I have the same thermostat. Would you be able to provide some guidance on where to start please

**Comment by jmgutierrezc1** [comment]
what pin has th temp sensor?

**Reply by hillaliy** [reply]
If you mean PGIO? There isn't!
It's work with serial commands

**Reply by jmgutierrezc1** [reply]
I see thanks. I am not familiar with UART communication with the MCU

**Comment by teotulip** [comment]
In the new thermostats as we know, the new TYWE3S chip is mounted which I replaced by unsoldering it with an esp8266 and loaded the HAA firmware
Once ready I loaded the script reported by @ravensystem here: https://github.com/RavenSystem/esp-homekit-devices/discussions/1834#discussioncomment-6078025
It works and takes the command correctly from Homekit, but the temperature sensor remains at -99°, even the external sensor, which I will not use, remains at -99°
What's wrong? can the UART commands coming from the device be changed?
![Screenshot 2024-11-12 alle 22 55 11](https://github.com/user-attachments/assets/c07b95e2-1249-4495-8e84-a846b395ee20)
![Screenshot 2024-12-08 alle 12 18 16](https://github.com/user-attachments/assets/fd6e0795-6f6a-4953-8a13-619724ca4074)

**Comment by hillaliy** [comment]
Moes changed the chip to  WB3S (BK7231T). You will be able to send command from HomeKit but HomeKit won't get info from Device. If you turn on from the button you don't get the status on HomeKit. No solution yet.

**Reply by teotulip** [reply]
> Moes changed the chip to WB3S (BK7231T). You will be able to send command from HomeKit but HomeKit won't get info from Device. If you turn on from the button you don't get the status on HomeKit. No solution yet.

You're right, I didn't notice it was one-way.
Too bad.

**Comment by TriderG75** [comment]
Are you on Discord too?Inviato da iPhoneIl giorno 16 apr 2025, alle ore 04:44, jmgutierrezc1 ***@***.***> ha scritto:﻿
Not clear what is happening. I replaced the original chip with an esp8266 12. Not sure that may cause the lack of communication but sure that it is not working even after re-pairing it. Thanks for your help

**Comment by jmgutierrezc1** [comment]
Not working….thanks for the message.Enviado desde mi iPhoneEl 5 may 2025, a la(s) 10:38 a.m., TriderG75 ***@***.***> escribió:﻿
Still not working? I'm used it for a few day and it's ok

**Reply by TriderG75** [reply]
Find me in the haa channel on discord so we can speak in Spanish if you want

**Reply by jmgutierrezc1** [reply]
Gracias, I am no familiar with the channel in discord.

**Reply by TriderG75** [reply]
Here: https://discord.com/invite/v8hyxj2

**Comment by jmgutierrezc1** [comment]
Nunca he entrado a discord pero te busco.Gracias Enviado desde mi iPhoneEl 7 may 2025, a la(s) 11:24 p.m., TriderG75 ***@***.***> escribió:﻿
Find me in the haa channel on discord so we can speak in Spanish if you want

**Comment by pilot1981** [comment]
Hi,
I'm not so good with soldering tools to exchange TUYA chipset...anyone found a commercial thermostat with ESP chip?

**Reply by mariusvasss** [reply]
Try Meross but check the amps.
